erpnext/stock/doctype/packed_item/packed_item.py

```diff
@@ -52,28 +51,35 @@ def update_packing_list_item(doc, packing_item_code, qty, main_item_row, descrip
 	pi.uom = item.stock_uom
 	pi.qty = flt(qty)
 	pi.description = description
-	if not pi.warehouse:
+	if not pi.warehouse and not doc.amended_from:
 		pi.warehouse = (main_item_row.warehouse if ((doc.get('is_pos')
 			or not item.default_warehouse) and main_item_row.warehouse) else item.default_warehouse)
-
-	if not pi.batch_no:
+	if not pi.batch_no and not doc.amended_from:
 		pi.batch_no = cstr(main_item_row.get("batch_no"))
 	if not pi.target_warehouse:
 		pi.target_warehouse = main_item_row.get("target_warehouse")
 	bin = get_bin_qty(packing_item_code, pi.warehouse)
 	pi.actual_qty = flt(bin.get("actual_qty"))
 	pi.projected_qty = flt(bin.get("projected_qty"))
 
+	if old_packed_items_map:
+		pi.batch_no = old_packed_items_map[packing_item_code]["batch_no"]
+		pi.serial_no = old_packed_items_map[packing_item_code]["serial_no"]
+		pi.warehouse = old_packed_items_map[packing_item_code]["warehouse"]
+
 def make_packing_list(doc):
 	"""make packing list for Product Bundle item"""
-
 	if doc.get("_action") and doc._action == "update_after_submit": return
+	if doc.amended_from:
+		old_packed_items_map = get_old_packed_item_details(doc.packed_items)
+	else:
+		old_packed_items_map = False
 
 	parent_items = []
 	for d in doc.get("items"):
 		if frappe.db.get_value("Product Bundle", {"new_item_code": d.item_code}):
 			for i in get_product_bundle_items(d.item_code):
-				update_packing_list_item(doc, i.item_code, flt(i.qty)*flt(d.stock_qty), d, i.description)
+				update_packing_list_item(doc, i.item_code, flt(i.qty)*flt(d.stock_qty), d, i.description, old_packed_items_map)
```

Comment:
why it should be passed as an argument? it can be called directly in update_packing_list_item